lib/magma/actions/model_update_actions.rb

```diff
@@ -1,26 +1,71 @@
+require_relative 'base_action'
 require_relative 'update_attribute'
+require_relative 'add_attribute'
 
 class Magma
   class ModelUpdateActions
+    class FailedActionError < StandardError; end
+
     def self.build(project_name, actions_list)
       new(project_name, actions_list)
     end
 
     def perform
-      valid? && @actions.all?(&:perform)
+      return false unless valid?
+
+      Magma.instance.db.transaction do
+        raise FailedActionError unless @actions.all?(&:perform)
+        update_model_tables
+        true
+      end
+    rescue
+      @actions.each(&:rollback)
+      false
     end
 
     def errors
-      @errors + @actions.flat_map(&:errors)
+      @errors.map(&:to_h) + @actions.flat_map(&:errors)
     end
 
     private
 
+    def update_model_tables
+      migrations = @project.migrations
+      return if migrations.all?(&:empty?)
+
+      sequel_migration = eval("
+        Sequel.migration do
+          up do
+            #{migrations.map(&:to_s).join("\n")}
+          end
+        end
+      ")
+
+      sequel_migration.apply(Magma.instance.db, :up)
+      restart_server
+    end
+
+    def restart_server
+      return if Magma.instance.test?
+      Process.kill("USR2", Magma.instance.server_pid)
```

Comment:
`USR2` is normal/hot restart. From the puma docs

>A hot restart means that no requests will be lost while deploying your new code, since the server socket is kept open between restarts.
>
> But beware, hot restart does not mean that the incoming requests won’t hang for multiple seconds while your new code has not fully deployed. If you need a zero downtime and zero hanging requests deploy, you must use phased restart.

https://github.com/puma/puma/blob/e2b419341153ff2206e1cb84fca1663639d0ef30/docs/restart.md